bgpd/bgpd.c

```diff
@@ -3815,9 +3816,9 @@ void peer_change_action(struct peer *peer, afi_t afi, safi_t safi,
 				BGP_NOTIFY_CEASE_CONFIG_CHANGE);
 	} else if (type == peer_change_reset_in) {
 		if (CHECK_FLAG(peer->cap, PEER_CAP_REFRESH_OLD_RCV)
-		    || CHECK_FLAG(peer->cap, PEER_CAP_REFRESH_NEW_RCV))
-			bgp_route_refresh_send(peer, afi, safi, 0, 0, 0);
-		else {
+		    || CHECK_FLAG(peer->cap, PEER_CAP_REFRESH_NEW_RCV)) {
+			bgp_route_refresh_send(peer, afi, safi, 0, 0, 0, 0);
```

Comment:
Sending of route refresh capability for respective AFI-SAFI should only be done if that AFI-SAFI is negotiated with this peer.